futures-test/src/future/assert_unmoved.rs

```diff
@@ -9,11 +9,14 @@ use std::thread::panicking;
 /// Combinator for the
 /// [`FutureTestExt::assert_unmoved`](super::FutureTestExt::assert_unmoved)
 /// method.
+#[pin_project(PinnedDrop)]
 #[derive(Debug, Clone)]
 #[must_use = "futures do nothing unless you `.await` or poll them"]
 pub struct AssertUnmoved<Fut> {
+    #[pin]
     future: Fut,
     this_ptr: *const AssertUnmoved<Fut>,
+    #[pin]
     _pinned: PhantomPinned,
```

Comment:
Oh, I just added [an option for this in pin-project.](https://docs.rs/pin-project/0.4.14/pin_project/attr.pin_project.html#unpin)